jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -15,194 +15,161 @@
+                        default -> log.error("Failed to decode ASN.1 data");
+                    }
+                    bis = removeNextNewLineCharacter(bis);
+                } catch (Exception e) {
+                    throw new LogFileToAsn1CodecPublisherException("Error parsing or publishing data.", e);
+                }
+            } while (status == ParserStatus.COMPLETE);
+        }
+
+        return dataList;
+    }
+
+    private void addDataToList(List<OdeData> dataList) {
+
+        OdeData odeData;
+
+        OdeMsgPayload payload;
+        OdeLogMetadata metadata;
+        if (isDriverAlertRecord()) {
+            payload = new OdeDriverAlertPayload(((DriverAlertFileParser) fileParser).getAlert());
+            metadata = new OdeLogMetadata(payload);
+            odeData = new OdeDriverAlertData(metadata, payload);
+        } else if (isBsmRecord()) {
+            payload = new OdeAsn1Payload(fileParser.getPayloadParser().getPayload());
+            metadata = new OdeBsmMetadata(payload);
+            odeData = new OdeAsn1Data(metadata, payload);
+        } else if (isSpatRecord()) {
+            payload = new OdeAsn1Payload(fileParser.getPayloadParser().getPayload());
+            metadata = new OdeSpatMetadata(payload);
+            odeData = new OdeAsn1Data(metadata, payload);
+        } else {
+            payload = new OdeAsn1Payload(fileParser.getPayloadParser().getPayload());
+            metadata = new OdeLogMetadata(payload);
+            odeData = new OdeAsn1Data(metadata, payload);
+        }
+        fileParser.updateMetadata(metadata);
+
+        dataList.add(odeData);
+    }
+
+    public boolean isDriverAlertRecord() {
+        return fileParser instanceof DriverAlertFileParser;
+    }
+
+    public boolean isBsmRecord() {
+        return fileParser instanceof BsmLogFileParser || (fileParser instanceof RxMsgFileParser
+                && ((RxMsgFileParser) fileParser).getRxSource() == RxSource.RV);
+    }
+
+    public boolean isSpatRecord() {
+        return fileParser instanceof SpatLogFileParser;
+    }
+
+    private void publishList(List<OdeData> dataList) {
+        serialId.setBundleSize(dataList.size());
+
+        for (OdeData odeData : dataList) {
+            OdeLogMetadata msgMetadata = (OdeLogMetadata) odeData.getMetadata();
+            OdeMsgPayload msgPayload = odeData.getPayload();
+            msgMetadata.setSerialId(serialId);
+
+            if (isDriverAlertRecord()) {
+                publisher.publish(jsonTopics.getDriverAlert(), JsonUtils.toJson(odeData, false));
+            } else if (isBsmRecord()) {
+                publisher.publish(rawEncodedJsonTopics.getBsm(), JsonUtils.toJson(odeData, false));
+            } else if (isSpatRecord()) {
+                publisher.publish(rawEncodedJsonTopics.getSpat(), JsonUtils.toJson(odeData, false));
+            } else {
+                // Determine the message type (MAP, TIM, SSM, SRM, or PSM)
```

Comment:
I just removed it in https://github.com/CDOT-CV/jpo-ode/pull/112/commits/f518c01193b4305ec34bd867ae4485e444c9b633 because it's an inaccurate comment telling us what the function call on the next line already tells us